### DISCUSSION TITLE
Port to foobar2000 as a visualization component

### DISCUSSION THREAD

**Discussion by TF3RDL** [hvianna/audioMotion-analyzer#50] [Ideas]
It would be nice to port this as foobar2000 v2.0 x64 component because Musical Spectrum (a nice alternative to default spectrum) hasn't been updated to support 64-bit versions of fb2k v2.0 and newer, and there is no alternatives for that ([foo_fft_analyzer](https://fanon.fandom.com/wiki/FFT_Analyzer_and_CQT_Analyzer) doesn't count since it is made-up as I said multiple times).

However, it requires re-writing the entire thing to C++ and use foobar2000 SDK as Spider Monkey Panel doesn't support visualizations and Canvas APIs.

**Comment by hvianna** [comment]
As a foobar2000 user myself, I'd **love** to see this happen! 😍 I currently use [foo_musical_spectrum](https://wiki.hydrogenaud.io/index.php?title=Foobar2000%3AComponents%2FMusical_Spectrum_%28foo_musical_spectrum%29) for visualization.

Since I have no experience with C/C++ I would need to learn a lot, but I haven't been able to dedicate any time to work on audioMotion for these past several months.

But if anyone is up to the task, I'd be more than happy to help as I can.

**Reply by TF3RDL** [reply]
Same for me though I use the 64-bit version of fb2k v2.0 and I've even managed to recreate foo_uie_vis_channel_spectrum on [my first CodePen audio visualizer project](https://codepen.io/TF3RDL/pen/NWBYeaE) as a part of mockups of supposed Wamoc's foobar2000 v2.0 components (which is a part of a [discussion about recreating the functionality of fb2k components](https://hydrogenaud.io/index.php/topic,124330.0.html) that doesn't even exist in the first place) and last time, I made the [mockup of foo_cqt_analyzer](https://codepen.io/TF3RDL/pen/poQJwRW) and [its underlying algorithm](https://codepen.io/TF3RDL/pen/XWyRBoo)

I also pointed out that foobar2000 visualizations actually captures samples ahead of actual playback unlike Web Audio API visualizers (barring from MusicVid.org and Vizzy.io), but it is still possible to implement sDFT algorithm for real-time CQT/VQT

**Reply by TF3RDL** [reply]
@stuerp made a [Musical Spectrum remake](https://www.foobar2000.org/components/view/foo_vis_spectrum_analyzer) for foobar2000 x64 (which is the first step for making audioMotion-analyzer into foobar2000 and making my concept components especially [foo_fft_analyzer and foo_cqt_analyzer](https://fanon.fandom.com/wiki/FFT_Analyzer_and_CQT_analyzer) a reality), but right now, this component is closed-source until he is confident enough to release the source code

So it is a good idea for @hvianna to collaborate with the developer of Musical Spectrum remake (foo_vis_spectrum_analyzer) once the source code for this aforementioned component is released?
